### ISSUE TITLE
Unification: Multiplayer Scoring

### ISSUE THREAD

**Issue by Matt1360** [dxx-rebirth/dxx-rebirth#48]
I'll be looking into it as I test upcoming tracker stuff - this is simply a "don't forget" note.

Player A hosts, and B joins.

A kills B.
A sees score: A 1, B 0.  Also, scoring appears as B on top with 0 points, and A on bottom with 1 point.
B sees score: A -1, B 0.  Also, scoring appears as A on top with -1 points, B on bottom with 0 points.
A's console reads that B joined game, and that "You killed B", as expected.
B's console reads "A committed suicide!" and not even a note was left :(

B now kills A.
A sees score: A 1, B 1.  Score order didn't change, B on top of A, but tied.
B sees score: A -1, B 1.  Score order didn't change, A on top of B, while A has a lesser score.
A's console reads that "B killed you!", as expected.
B's console reads "You killed B!", as expected.

So, [pretty broken](http://i.imgur.com/6NfmQ.jpg).  I'll try to look into this soon.

**Comment by Matt1360** [comment]
Interesting.  If B kills himself (including while A is waiting to respawn), B's console reads that "You killed A!", and B's kills and score increases by one, and *deaths do not*.  A's console, and score read correctly.  [Welp](http://i.imgur.com/g2lQx.gif).

**Comment by zicodxx** [comment]
The score calculation in Multiplayer is done in the functions multi_send_kill() and multi_do_kill() depending on who killed who.
So it is safe to assume the issue happens there.

In the past I changed this function so only host computes the actual kill to make sure scores stay the same among all players regardless of packet loss/delay. This change was thoroughly tested so I assume the bug happened after that. But this could explain why the effects differ for A (Host) and B (Client).

What I can imagine is two scenarios:
a) Something in the function changed and the parsing of the kill packet is off by X bytes, messing up the kill  matrix
b) The change of the data type of Player_num and related variables from sbyte to playernum_t could maybe also cause this

I am not certain which one may it be but that could be a place to start looking for the issue.

**Comment by vLKp** [comment]
This should be fixed by 5d6e2c16c609d8994cd260a91f15223e92c195c8.  The problem was caused by switching `multi_do_kill` to receive player number as an argument.  Many multiplayer functions act on the sending player, but the sending player's number is reiterated in the message.  I had changed some of those functions to rely on the player number from the control header.  In the case of `multi_do_kill`, the player specified in the control header is the host, but the dying player may not be.
